src/node/tendermint_processes.ml

```diff
@@ -0,0 +1,367 @@
+  (* If we start a round > 0 and already have a valid/locked value, we update their hash
+     with the correct round *)
+  (if consensus_state.valid_value <> nil then
+     let new_value = update_value consensus_state.valid_value round in
+     consensus_state.valid_value <- new_value);
+  (if consensus_state.locked_value <> nil then
+     let new_value = update_value consensus_state.locked_value round in
+     consensus_state.locked_value <- new_value);
+  let return_action =
+    if i_am_proposer global_state height round then
+      if consensus_state.valid_value <> nil then
+        Some
+          (Broadcast
+             (ProposalOP
+                ( consensus_state.height,
+                  consensus_state.round,
+                  consensus_state.valid_value,
+                  consensus_state.valid_round )))
+      else
+        Some
+          (Broadcast
+             (ProposalOP
+                ( consensus_state.height,
+                  consensus_state.round,
+                  (* FIXME: remove depdendency to full global_state *)
+                  !produce_value global_state,
+                  consensus_state.valid_round )))
+    else
+      Some
+        (Schedule
+           (Clock.make proposal_timeout Proposal height round
+              (on_timeout_propose height round))) in
+  return_action
+
+let on_timeout_precommit (height : height) (round : round)
+    (consensus_state : consensus_state) (msg_log : input_log) _dlog
+    _global_state =
+  let do_something consensus_state =
+    if height = consensus_state.height && round = consensus_state.round then
+      RestartTendermint (height, round + 1)
+    else
+      DoNothing in
+  if contains_timeout msg_log height Precommit then
+    Some (do_something consensus_state)
+  else
+    None
+
+let response_to_proposal (height : height) (round : round)
+    (consensus_state : consensus_state) (msg_log : input_log) _dlog global_state
+    =
+  let _height = consensus_state.height in
+  let step = consensus_state.step in
+  let _round = consensus_state.round in
+  let do_something (consensus_state : consensus_state) (found_set : MySet.t) =
+    let value, _ = MySet.choose found_set in
+    consensus_state.step <- Prevote;
+    if
+      is_valid global_state value
+      && (consensus_state.locked_round = -1
+         || consensus_state.locked_value = value)
+    then
+      Broadcast (PrevoteOP (height, round, repr_of_value value))
+    else
+      Broadcast (PrevoteOP (height, round, nil)) in
+  let found_set =
+    select_proposal_process_round_matching_proposer msg_log consensus_state
+      global_state in
+  let extract_valid_set (found_set : MySet.t) =
+    MySet.filter (fun (_, r) -> r = round) found_set in
+  let valid_set = extract_valid_set found_set in
+  if valid_set = MySet.empty || step <> Proposal then
+    None
+  else
+    Some (do_something consensus_state valid_set)
+
+let precommit_failed_previous_round (_height : height) (_round : round)
```

Comment:
_round _height _dlog

Reply:
Unused here, however, we need all the Tendermint process functions to have the same signature.